### PR TITLE
Enhancements and Alignments in Errors

### LINKED ISSUE #321: Enhancements and Alignments in Errors
**Problem description**
Two enhancements are proposed inder this issue:
- Normalize mandatory errors that are common to every CAMARA API.
- Define specific Errors for `phoneNumber` concept in the same fashion as done for `device` in MetaRelease Fall 24 for [device error responses](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#62-error-responses---device-object) 

**Possible evolution**

1) Normalize Mandatory Error status
- Add within section `6.1 Standardized use of CAMARA error responses` the following information:

Mandatory Errors to be documented in CAMARA API Spec YAML are the following:
- For event notifications flow, the ones defined in [notification-as-cloud-event.yaml](https://github.com/camaraproject/Commonalities/blob/main/artifacts/notification-as-cloud-event.yaml)
- For event subscriptions APIs, the ones defined in [event-subscription-template.yaml](https://github.com/camaraproject/Commonalities/blob/main/artifacts/camara-cloudevents/event-subscription-template.yaml)
- For the rest of APIs:
> Error status 401
> Error status 403
> Error status 429 TOO_MANY_REQUESTS (For rate limit control)

> NOTE: The rest of Error status defined in section 6.1. will apply depending on specific considerations within the given WG.

2) Define specific Errors for `phoneNumber` concept

Add a new section `6.3 Error Responses - phoneNumber` to detail error scenarios for phoneNumber

| **Case #** | **Description**                                                            | **Error status** |         **Error code**         | **Message example**                                      |
|:----------:|:---------------------------------------------------------------------------|:----------------:|:------------------------------:|:---------------------------------------------------------|
|     0      | The request body does not comply with the schema                           |       400        |        INVALID_ARGUMENT        | Request body does not comply with the schema.           |
|     1      | The phoneNumber provided is not supported by the implementation |       422        | UNSUPPORTED_PHONE_NUMBER | phoneNumber is required.                                 |
|     2      | phoneNumber is not found.                    |       404        |        PHONE_NUMBER_NOT_FOUND        | phoneNumber not found.                             |  
|     3      | Invalid access token context                                               |       403        |     INVALID_TOKEN_CONTEXT      | phoneNumber is not consistent with access token. |
|     4      | Service not applicable to the phoneNumber                                    |       422        |     PHONE_NUMBER_NOT_APPLICABLE      | The service is not available for the provided phoneNumber.    |
|     5      | The phone number is not included in the request and the phone number information cannot be derived from the 3-legged access token |       422        |     UNIDENTIFIABLE_PHONE_NUMBER      | The phoneNumber cannot be identified. |
|     6      | The phoneNumber is unnecesarily included in a 3-legged scenario where the subscription is identified from the 3-legged access token |      422        |      UNNECESSARY_PHONE_NUMBER        |  The phoneNumber is unnecesarily provided

NOTE: New exceptions would be added to `CAMARA_common.yaml` for reference

**Alternative solution**

**Additional context**

There are CAMARA APIs where phoneNumber represents the end's user subscription (e.g SIM-SWAP, Number Verification, Carrier Billing, KYC family,...). Therefore having specific errors related to the concept of phoneNumber is the natural way to address error scenarios in a fully semantic and user-friendly way

Proposed as a child issue of #306

### PR DESCRIPTION
**Pull request by PedroDiez**
#### What type of PR is this?

* enhancement

#### What this PR does / why we need it:

This PR is to address agreements within #321
1) Alignment in Mandatory Error `status` to be documented in API Spec YAML
2) Alignment in Error codes regarding "API Subject" identifier (device or phoneNumber) in API Design Guidelines Section 6.2. Aligned with #324 

#### Which issue(s) this PR fixes:

Fixes #321

#### Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

Changes in error codes to be used in API defintions

#### Special notes for reviewers:

Ready for review

#### Changelog input

```
Commonalities aligment regarding mandatory error `status` in API Spec YAML and common terminology for "API Subject" identifier codes. 

```

#### Additional documentation 

N/A

### CODE REVIEW

**Review by patrice-conil — DISMISSED** [review]
LGTM

File: documentation/API-design-guidelines.md [hunk 1]
```diff
@@ -777,25 +777,37 @@ In the following, we elaborate on the existing client errors. In particular, we
...
 
 > _NOTE 2: A {{SPECIFIC_CODE}}, unless it may have traversal scope (i.e. re-usable among different APIs), SHALL follow this scheme for a specific API: {{API_NAME}}.{{SPECIFIC_CODE}}_
 
-### 6.2 Error Responses - Device Object
 
-This section is focused in the guidelines about error responses around the concept of `device` object.
+**Mandatory Errors** to be **documented in CAMARA API Spec YAML** are the following:
+
+- For event notifications flow, the ones defined in [notification-as-cloud-event.yaml](/artifacts/notification-as-cloud-event.yaml)
+- For event subscriptions APIs, the ones defined in [event-subscription-template.yaml](/artifacts/camara-cloudevents/event-subscription-template.yaml)
```
Comment (rartych) [inline documentation/API-design-guidelines.md:788]: I would prefer to include all normative requirements in the guidelines and use artifacts as illustrative files or templates.
It looks that we have respective requirements in chapter 12:
```suggestion
**Mandatory Errors** to be **documented in CAMARA API Spec YAML** are the following:

- For event subscriptions APIs, the ones defined in [12.1 Subscription](#error-definition-for-resource-based-explicit-subscription)
- For event notifications flow, the ones defined in [12.2 Event notification](#error-definition-for-event-notification)
```

Currently these points in Design Guidelines are not changed in PR #313, 
I propose to introduce these changes within this PR #329 after merging PR #313 

WDYT @bigludo7 @PedroDiez  ?

File: documentation/API-design-guidelines.md [hunk 2]
```diff
@@ -777,25 +777,37 @@ In the following, we elaborate on the existing client errors. In particular, we
...
+- For the rest of APIs:
+  - Error status 401
+  - Error status 403
+  - Error status 429 TOO_MANY_REQUESTS (For rate limit control)
+
+NOTE: The rest of Error status defined in section 6.1. will be documented depending on specific considerations within a given WG.
```
Comment (rartych) [inline documentation/API-design-guidelines.md:794]: Rephrasing and avoiding "WG" (it is rather Sub project or sandbox project working on API definition)

```suggestion
NOTE: The remaining Error statuses defined in section 6.1 will be documented based on specific considerations for the given API.
```

File: documentation/API-design-guidelines.md (hunk 1 above)
Comment (PedroDiez) [reply documentation/API-design-guidelines.md:788]: It is fine, to be adressed after PR#313

**Comment by PedroDiez** [comment]
Working on this, i will advise you @rartych, @patrice-conil

**Comment by PedroDiez** [comment]
Please, you can check: latest commit https://github.com/camaraproject/Commonalities/pull/329/commits/9365440860c22073e56af671983fbbc18b462cd4

Hope have considered suitably your comments Rafal:

- Haave aligned with PR #324
- Also aligned 422 errors info between API design guidelines and `CAMARA_common.yaml`

**Comment by PedroDiez** [comment]
NOTE: Pending to review and align section 6.1. Please do not review yet

**Comment by PedroDiez** [comment]
**02/DIC**: Ready for Review

**Review by patrice-conil — APPROVED** [review]
LGTM

**Review by rartych — APPROVED** [review]
LGTM

**Pull request merged**